virtinst/virtxml.py

```diff
@@ -63,37 +63,111 @@ def defined_xml_is_unchanged(conn, domain, original_xml):
 # Action parsing #
 ##################
 
-def check_action_collision(options):
-    actions = ["edit", "add-device", "remove-device", "build-xml"]
+class Action:
+    """
+    Helper class tracking one pair of
+        XML ACTION (ex. --edit) and
+        XML OPTION (ex. --disk)
+    """
+    def __init__(self, action_string, selector, parserclass, parservalue):
+        # one of ["edit", "add-device", "remove-device", "build-xml"]
+        self.action_string = action_string
+        # ex. for `--edit 1` this is selector="1"
+        self.selector = selector
+        # ParserDisk, etc
+        self.parserclass = parserclass
+        # ex for `--disk path=/foo` this is "path=/foo"
+        self.parservalue = parservalue
+
+        self.is_edit = self.action_string == "edit"
+        self.is_add_device = self.action_string == "add-device"
+        self.is_remove_device = self.action_string == "remove-device"
+        self.is_build_xml = self.action_string == "build-xml"
```

Comment:
please make these as properties, rather than instance variables: this avoids the risk to modify them & to have a inconsistency between these and `action_string`